homeassistant/components/smlight/update.py

```diff
@@ -193,5 +192,13 @@ async def async_install(
             # block until update finished event received
             await self._finished_event.wait()
 
-            await self.coordinator.async_refresh()
+            # allow time for SLZB-06 to reboot before updating coordinator data
+            while (
+                self.coordinator.in_progress
+                and self.installed_version != self._firmware.ver
+            ):
+                await self.coordinator.async_refresh()
+                await asyncio.sleep(0)
+
+            self.coordinator.in_progress = False
```

Comment:
Shouldn't the `_attr_in_progress` also be set to False?

Reply:
> Shouldn't the `_attr_in_progress` also be set to False?

I dont set this explicitly since this gets set after `async_install` returns:
https://github.com/home-assistant/core/blob/5b2113c43da1d17dc09b47d23d5542d3860884cb/homeassistant/components/update/__init__.py#L505